src/infer_request.cc

```diff
@@ -393,7 +393,13 @@ Status
 InferenceRequest::Run(std::unique_ptr<InferenceRequest>& request)
 {
   RETURN_IF_ERROR(request->SetState(InferenceRequest::State::PENDING));
-  return request->model_raw_->Enqueue(request);
+  auto status = request->model_raw_->Enqueue(request);
+  if (!status.IsOk()) {
+    LOG_STATUS_ERROR(
+        request->SetState(InferenceRequest::State::RELEASED),
```

Comment:
Should the request release callback be called internally on failure? Calling request release should set the state and update accordingly

Reply:
As mentioned offline, I don't love the idea of setting the state to `RELEASED` when the request hasn't actually had it's release callbacks called. Is there another way?

Reply:
Ideally, the request should be set to pending state iff the Enqueue call was a success. However, there is an asynchronous interaction between the enqueue thread and and the backend threads that schedules the requests and moves it out from the queue. 
We should not invoke callbacks for the failed requests as the failed requests are owned by the caller.

Reply:
Added a new state for tracking the failed requests. Discussed offline that the logic might need some clean-up.